doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+    - [5.2.2 Show Commands](#522-show-commands)
+- [6 Serviceability and Debug](#6-serviceability-and-debug)
+- [7 Warm reboot Support](#7-warm-reboot-support)
+- [8 Unit Test Cases ](#8-unit-test-cases)
+
+# List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev  |    Date    |       Author        | Change Description                                           |
+|:--:|:--------:|:-----------------:|:------------------------------------------------------------:|
+| 0.1  |  |   Rajesh Sankaran   | Initial version                                              |
+| 0.2  |  | Rajesh Pukhraj Jain | Added Warm boot requirement/design details. Added CLI, Functional, Scalability & Warm boot test cases |
+| 0.3  |  |   Rajesh Sankaran   | Incorporated Review comments                                 |
+| 0.4  |  | Tapash Das, Hasan Naqvi | Added L3 VXLAN details |
+| 0.5  |  | Karthikeyan A | ARP and ND suppression |
+| 0.6  |  | Kishore Kunal | Added Fdbsycnd details |
+| 0.7  |  | Rajesh Sankaran | Click and SONiC CLI added |
+| 0.8 | | Hasan Naqvi | Linux kernel section and fdbsyncd testcases added |
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| **Term** | **Meaning**                               |
+| -------- | ----------------------------------------- |
+| BGP      | Border Gateway Protocol                   |
+| BUM      | Broadcast, Unknown unicast, and Multicast |
+| EVPN     | Ethernet Virtual Private Network          |
+| IMET     | Inclusive Multicast Ethernet Tag          |
+| IRB      | Integrated Routing and Bridging           |
+| L3GW     | Layer 3 Gateway                           |
+| NVO      | Network Virtualization Overlay            |
+| VNI      | VXLAN Network Identifier                  |
+| VRF      | Virtual Routing and Forwarding            |
+| VTEP     | VXLAN Tunnel End point                    |
+| VXLAN    | Virtual Extended LAN                      |
+
+# About this Manual
+
+This document provides general information about the EVPN VXLAN feature implementation based on RFC 7432 and 8365 in SONiC. 
+This feature is incremental to the SONiC.201911 release which is referred to as current implementation in this document.
+
+
+# 1 Introduction and Scope
+
+This document describes the Functionality and High level design of the EVPN VXLAN feature.
+
+The following are the benefits of an EVPN approach for VXLAN.
+
+ - Standards based and Interoperable solution.
+ - Auto discovery of remote VTEPs, Auto provisioning of tunnels and VLANs over VXLAN tunnels.
+ - Support for L2 and L3 VPN services.
+ - Allows for dual homing support.
+ - Control plane MAC learning and ARP suppression leads to reduced flooding over an EVPN network.
+ - Eases planning and configuration when supporting downstream assigned VNI for DCI usecases.
+ - Aids in VM mobility.
+
+In the current implementation, there is support for L3 forwarding over VXLAN and EVPN Type 5 route support based on VNET constructs. 
```

Comment:
This VNET based EVPN type 5 only works with some ASIC SAI, but not all, right?

Reply:
This statement was from a SONiC perspective.  Compliance to SAI and Integration with SONiC per ASIC is outside the scope of this document.